File: ts/project/.sdk/tm/java/test/NetsimTest.java
package JAVAPACKAGE.sdktest;

// Network-behaviour simulation over the offline mock transport. The `test`
// feature accepts an optional `net` config so unit tests can exercise slow,
// failing and offline conditions without a live server. These checks drive
// the transport through direct(), which needs no entity, so they run for
// every generated SDK regardless of its API shape. Mirrors
// tm/go/test/netsim_test.go.

import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertNotNull;
import static org.junit.jupiter.api.Assertions.assertTrue;

import static JAVAPACKAGE.sdktest.FeatureHarness.fhMap;

import java.util.Map;

import org.junit.jupiter.api.Test;

import JAVAPACKAGE.core.ProjectNameSDK;

public class NetsimTest {

  @Test
  public void offlineSimulationFailsRequest() {
    ProjectNameSDK client = ProjectNameSDK.testSDK(fhMap(
        "net", fhMap("offline", true)), null);
    Map<String, Object> res = client.direct(fhMap("path", "/ping"));
    assertEquals(false, res.get("ok"), "offline network must fail the call: " + res);
  }

  @Test
  public void failstatusSimulationSurfacesStatus() {
    ProjectNameSDK client = ProjectNameSDK.testSDK(fhMap(
        "net", fhMap("failTimes", 1, "failStatus", 503)), null);
    Map<String, Object> res = client.direct(fhMap("path", "/ping"));
    assertEquals(false, res.get("ok"), "expected failed call: " + res);
    assertEquals(503, res.get("status"), "expected simulated 503");
  }

  @Test
  public void latencySimulationDelaysRequest() {
    int delay = 60;
    ProjectNameSDK client = ProjectNameSDK.testSDK(fhMap(
        "net", fhMap("latency", delay)), null);
    long start = System.currentTimeMillis();
    client.direct(fhMap("path", "/ping"));
    long elapsed = System.currentTimeMillis() - start;
    // Generous lower bound to stay robust on slow CI.
    assertTrue(elapsed >= delay - 25,
        "expected >= " + (delay - 25) + "ms latency, got " + elapsed + "ms");
  }

  @Test
  public void plainTestSdkWorksWithoutNet() {
    ProjectNameSDK client = ProjectNameSDK.testSDK();
    assertNotNull(client, "expected a client");
  }
}

File: ts/project/.sdk/tm/kotlin/test/NetsimTest.kt
package KOTLINPACKAGE.sdktest

// Network-behaviour simulation over the offline mock transport. The `test`
// feature accepts an optional `net` config so unit tests can exercise slow,
// failing and offline conditions without a live server. These checks drive
// the transport through direct(), which needs no entity, so they run for
// every generated SDK regardless of its API shape.

import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Assertions.assertNotNull
import org.junit.jupiter.api.Assertions.assertTrue
import org.junit.jupiter.api.Test

import KOTLINPACKAGE.core.ProjectNameSDK
import KOTLINPACKAGE.sdktest.FeatureHarness.fhMap

class NetsimTest {

  @Test
  fun offlineSimulationFailsRequest() {
    val client = ProjectNameSDK.testSDK(fhMap("net", fhMap("offline", true)), null)
    val res = client.direct(fhMap("path", "/ping"))
    assertEquals(false, res["ok"], "offline network must fail the call: $res")
  }

  @Test
  fun failstatusSimulationSurfacesStatus() {
    val client = ProjectNameSDK.testSDK(fhMap("net", fhMap("failTimes", 1, "failStatus", 503)), null)
    val res = client.direct(fhMap("path", "/ping"))
    assertEquals(false, res["ok"], "expected failed call: $res")
    assertEquals(503, res["status"], "expected simulated 503")
  }

  @Test
  fun latencySimulationDelaysRequest() {
    val delay = 60
    val client = ProjectNameSDK.testSDK(fhMap("net", fhMap("latency", delay)), null)
    val start = System.currentTimeMillis()
    client.direct(fhMap("path", "/ping"))
    val elapsed = System.currentTimeMillis() - start
    // Generous lower bound to stay robust on slow CI.
    assertTrue(elapsed >= delay - 25, "expected >= ${delay - 25}ms latency, got ${elapsed}ms")
  }

  @Test
  fun plainTestSdkWorksWithoutNet() {
    val client = ProjectNameSDK.testSDK()
    assertNotNull(client, "expected a client")
  }
}
